tests/integration/golang/mlflow/experiment/search_test.go

```diff
@@ -1,3 +1,255 @@
+	)
+	assert.Nil(s.T(), err)
+
+	return resp
+}
+
+func (s *SearchExperimentsTestSuite) Test_Ok() {
+	// 1. prepare database with test data.
+	experiments := []models.Experiment{
+		{
+			Name: "Test Experiment 1",
+			Tags: []models.ExperimentTag{
+				{
+					Key:   "key2",
+					Value: "value2",
+				},
+			},
+
+			LifecycleStage: models.LifecycleStageActive,
+		},
+		{
+			Name: "Test Experiment 2",
+			Tags: []models.ExperimentTag{
+				{
+					Key:   "key1",
+					Value: "value1",
+				},
+			},
+			LifecycleStage: models.LifecycleStageActive,
+		},
+		{
+			Name: "Test Experiment 3",
+			Tags: []models.ExperimentTag{
+				{
+					Key:   "key3",
+					Value: "value3",
+				},
+			},
+			LifecycleStage: models.LifecycleStageActive,
+		},
+		{
+			Name: "Test Experiment 4",
+			Tags: []models.ExperimentTag{
+				{
+					Key:   "key4",
+					Value: "value4",
+				},
+			},
+			LifecycleStage: models.LifecycleStageActive,
+		},
+		{
+			Name:           "Test Experiment 5",
+			Tags:           nil,
+			LifecycleStage: models.LifecycleStageActive,
+		},
+		{
+			Name:           "Test Experiment 6",
+			Tags:           nil,
+			LifecycleStage: models.LifecycleStageDeleted,
+		},
+	}
+	for _, ex := range experiments {
+		_, err := s.fixtures.CreateTestExperiment(context.Background(), &models.Experiment{
+			Name: ex.Name,
+			Tags: ex.Tags,
+			CreationTime: sql.NullInt64{
+				Int64: time.Now().UTC().UnixMilli(),
+				Valid: true,
+			},
+			LastUpdateTime: sql.NullInt64{
+				Int64: time.Now().UTC().UnixMilli(),
+				Valid: true,
+			},
+			LifecycleStage:   ex.LifecycleStage,
+			ArtifactLocation: "/artifact/location",
+		})
+		assert.Nil(s.T(), err)
+	}
+	defer func() {
+		assert.Nil(s.T(), s.fixtures.UnloadFixtures())
+	}()
+
+	// Test Filter
+	resp := GetSearchExperimentsResponse(s, request.SearchExperimentsRequest{
+		Filter: "attribute.name != 'Test Experiment 5'",
+	})
+
+	assert.ElementsMatch(s.T(), []string{"Test Experiment 1", "Test Experiment 2", "Test Experiment 3", "Test Experiment 4", "Default"}, GetExperimentNames(resp.Experiments))
+
+	//Test ViewType
+	resp = GetSearchExperimentsResponse(s, request.SearchExperimentsRequest{
+		ViewType: request.ViewTypeDeletedOnly,
+	})
+
+	assert.ElementsMatch(s.T(), []string{"Test Experiment 6"}, GetExperimentNames(resp.Experiments))
+
+	//Test OrderBy
+	resp = GetSearchExperimentsResponse(s, request.SearchExperimentsRequest{
+		OrderBy: []string{"name ASC"},
+	})
+
+	assert.Equal(s.T(), []string{"Default", "Test Experiment 1", "Test Experiment 2", "Test Experiment 3", "Test Experiment 4", "Test Experiment 5"}, GetExperimentNames(resp.Experiments))
+
+	// Test MaxResults
+	resp = GetSearchExperimentsResponse(s, request.SearchExperimentsRequest{
+		OrderBy:    []string{"name ASC"},
+		MaxResults: 3,
+	})
+
+	assert.ElementsMatch(s.T(), []string{"Default", "Test Experiment 1", "Test Experiment 2"}, GetExperimentNames(resp.Experiments))
+}
+
+func (s *SearchExperimentsTestSuite) Test_Error() {
```

Comment:
Niceeee! we have tests for this pretty complex endpoint so it will be much safer to change it after your changes. Great work!